activerecord/lib/active_record/scoping/named.rb

```diff
@@ -30,13 +30,8 @@ def all
         end
 
         def default_scoped # :nodoc:
-          scope = build_default_scope
-
-          if scope
-            relation.spawn.merge!(scope)
-          else
-            relation
-          end
+          scope = relation
+          build_default_scope(scope) || scope
```

Comment:
Is it still worth avoiding `klass.all` entirely as per https://github.com/rails/rails/pull/29098#discussion_r118831126? When I measured it, it seemed to get us nearly half way towards the cached speed.